frontend/src/pages/signUp/SignUp.tsx

```diff
@@ -32,7 +32,7 @@ const SignUp = () => {
       setErrorMessage('');
       navigate('/login');
     } catch (e: any) {
-      if (e.message === 'Firebase: Error (auth/email-already-in-use).') {
+      if (e.code === 'auth/email-already-in-use') {
```

Comment:
Better :) But lets see if we can prevent using raw strings. It should be possible to use the constants here: https://firebase.google.com/docs/reference/js/auth#autherrorcodes

try importing like this:
`import { AuthErrorCodes } from 'firebase/auth'`

And then you should be able to use the constants for comparison:
`AuthErrorCodes.EMAIL_EXISTS`